parquet-protobuf/src/main/java/org/apache/parquet/proto/ProtoWriteSupport.java

```diff
@@ -445,10 +445,13 @@ final void writeField(Object value) {
 
     private void writeAllFields(MessageOrBuilder pb) {
       Descriptor messageDescriptor = pb.getDescriptorForType();
-      Descriptors.FileDescriptor.Syntax syntax =
-          messageDescriptor.getFile().getSyntax();
+      String syntax = messageDescriptor.getFile().toProto().getSyntax();
+      if ("editions".equals(syntax)) {
+        throw new UnsupportedOperationException("protocol buffers 'editions' not supported");
+      }
+      boolean isProto2 = !"proto3".equals(syntax);
```

Comment:
Wouldn't it be better to check the versions explicitly, and raise on anything else (for example `editions`, as above).
```suggestion
      boolean isProto2 = "proto2".equals(syntax);
```